client/packages/programs/src/JsonForms/components/IdGenerator.tsx

```diff
@@ -353,6 +353,7 @@ const UIComponent = (props: ControlProps) => {
       });
       // run till we find a valid id or till the generated id didn't change
       const finished = !(await validateId(options, nextId)) || id === nextId;
+      onChange(nextId);
```

Comment:
Looks good! could you please just move the onChange out of the while loop? it only needs to update the UI once a valid id found...

Reply:
[2cfc201](https://github.com/openmsupply/open-msupply/pull/1936/commits/2cfc2014554bd98567179db493c162a269faf8cf)